BDSP Egg Collector/Egg Hatcher Bots

zhekaiii opened the pull request:
It's been a while! Not sure if this is still in use since SwSh's hype got stolen by BDSP I guess, but I made an Egg Collector bot for BDSP. Kind of forgot what needs to be updated, but this bot is simple so hopefully I didn't miss anything out.

zhekaiii committed:
feat: Add BDSP Egg Collector Bot

zakthompson commented:
Wicked!  I've been meaning to come back to this and add some for BDSP, I just haven't had time yet.  Going to give this a test in the next couple days and then get it live!

zhekaiii commented:
I made this in a rush so I haven't had time to optimise the button presses, but currently the timing is quite smooth I would say? There were a few extra button presses that I had to add in to continue the dialogs. Those are the ones that i commented with ???. I thought I didn't put enough delay on the previous button presses, but I really couldn't get it to work without the extra presses. If you'd like you can try to optimise the presses, otherwise I think it works good enough to just use as it is. I'm on 1.1.2 in case there's a difference.

zhekaiii committed:
allow attempt limit for bdsp egg collector

zhekaiii committed:
add missing commands & optimize number of loops

zhekaiii committed:
EggHatcher for BDSP

zhekaiii commented:
Apologies if your email is getting spammed by GitHub notifications, but I optimized the spinning part of the Egg Collector, incorporated the use of the Counter app in the Pokétch, and made a BDSP Egg Hatcher as well.

Happy new year :)

zhekaiii committed:
Optimize Egg Hatcher loops, remove Counter from EggCollector

zhekaiii committed:
optimize button timings for EggHatcher

zhekaiii committed:
Optimize EggHatcher key press timings for non-full columns
- Empty box slots cause a delay before the game can register the next key press so certain key presses were not registered
